### ISSUE TITLE
RFE: Replace use of `pkg_resources` with `importlib.metadata`

### ISSUE THREAD

**Issue by kloczek** [pyparsing/pyparsing#391]
See discussions:
[astropy/astropy#11091](https://github.com/astropy/astropy/pull/11091)
[pypa/pip#7413](https://github.com/pypa/pip/issues/7413)

```console
[tkloczko@devel-g2v SRPMS]$ rpm -ql python-pyparsing | grep py$ | xargs grep pkg_resources
/usr/lib/python3.8/site-packages/pyparsing/diagram/__init__.py:from pkg_resources import resource_filename
```

**Comment by ptmcg** [comment]
I think the suggested changes do not support back to Py3.6, which for some reason I think is still worth supporting for pyparsing (I think maybe in the next release I'll announce that pyparsing 3.1 will be released later in 2022, and will drop support for 3.6.)

But we only explicitly use pkg_resources in one place, to locate and read the jinja2 template for the railroad diagrams. This template is only about 25 lines long, I don't see a lot of harm in just in inlining it in the railroad diagram code. Then we could just cut the use of pkg_resources and not need to come up with an equivalent.

**Comment by kloczek** [comment]
Hmm .. if it is some solution which may allow do the same without `pkg_resources` and would allow to detach `pyparsing` from `setuptools` (which part is `pkg_resources`) without breaking compatibility with Py3.6 I think that it would be worth to do that 🤔

**Comment by kloczek** [comment]
Nevertheless `seruptools` developer confirmed that trying find some replacement for use `pkg_resources` is kind of best longer term solution https://github.com/pypa/setuptools/issues/3279 (so it is nothing really-so-urgent for now)

**Comment by ptmcg** [comment]
Released in 3.0.9
